keyboards/handwired/tsubasa/keymaps/multifunc_encoder/keymap.c

```diff
@@ -0,0 +1,217 @@
+enum custom_keycodes {
+    ENC_MOD = SAFE_RANGE
+};
+
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [_BASE] = LAYOUT(
+         KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,                          KC_7,    KC_8,    KC_9,    KC_0, KC_MINS,  KC_EQL,
+       KC_TAB,    KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,                          KC_Y,    KC_U,    KC_I,    KC_O,    KC_P, KC_LBRC, KC_RBRC,
+      KC_LCTL,    KC_A,    KC_S,    KC_D,    KC_F,    KC_G,                          KC_H,    KC_J,    KC_K,    KC_L, KC_SCLN, KC_QUOT,
+      KC_LSFT,    KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,              ENC_MOD,    KC_N,    KC_M, KC_COMM,  KC_DOT, KC_SLSH,
+                         KC_ESC, KC_RGUI, KC_LALT,  KC_SPC, KC_LSFT,     KC_BSPC,  KC_ENT, MO(_FN), KC_BSLS,  KC_GRV
+    ),
+    [_FN] = LAYOUT(
+        KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,                         KC_F7,   KC_F8,   KC_F9,  KC_F10,  KC_F11,  KC_F12,
+      _______, XXXXXXX, KC_BTN1, KC_MS_U, KC_BTN2, RGB_TOG,                       KC_PGUP, XXXXXXX,   KC_UP, XXXXXXX, KC_PSCR, XXXXXXX, XXXXXXX,
+      _______, XXXXXXX, KC_MS_L, KC_MS_D, KC_MS_R, RGB_MOD,                       KC_PGDN, KC_LEFT, KC_DOWN, KC_RGHT, XXXXXXX, XXXXXXX,
+      _______, XXXXXXX, XXXXXXX, XXXXXXX, RGB_VAI, RGB_HUI,              _______, KC_BRIU, KC_BRID, XXXXXXX, XXXXXXX, XXXXXXX,
+      _______, _______, _______, _______, _______,                       KC_ACL1, KC_ACL2, _______, _______, _______
+    )
+};
+
+uint8_t mod_state;
+uint8_t encoder_mode = 0;
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    mod_state = get_mods();
+    switch (keycode) {
+
+        // make KC_LSHIFT + KC_BSPACE to KC_DEL
+        case KC_BSPC:
+        {
+            static bool delkey_registered;
+            if (record->event.pressed) {
+                if (mod_state & MOD_MASK_SHIFT) {
+                    del_mods(MOD_MASK_SHIFT);
+                    register_code(KC_DEL);
+                    delkey_registered = true;
+                    set_mods(mod_state);
+                    return false;
+                }
+            } else {
+                if (delkey_registered) {
+                    unregister_code(KC_DEL);
+                    delkey_registered = false;
+                    return false;
+                }
+            }
+            return true;
+        }
+        // push encoder and change encoder function
+        case ENC_MOD:
+        {
+            if (record->event.pressed) {
+                if (mod_state & MOD_MASK_SHIFT) {
+                    encoder_mode = (encoder_mode+1)%4;
+                    break;
+                } else {
+                    encoder_mode = (encoder_mode+3)%4;
+                    break;
+                }
+            } else {
+                // when released
+            }
+        }
+
+    }
+    return true;
+}
+
+bool should_process_keypress(void) { return true; }
```

Comment:
Should use [sync options](https://docs.qmk.fm/#/feature_split_keyboard?id=data-sync-options) if required
```suggestion
```

Reply:
Specifically, keycodes shouldn't be processed on the secondary side, at all.  

If you need functionality to show up there, then the data sync options cover that, including custom synced data. 

This includes stuff like custom modes.